File: can-place-flowers/can-place-flowers.java
class Solution {
    public boolean canPlaceFlowers(int[] flowerbed, int n) {
        if (n == 0) return true;
        
        for (int i = 0; i < flowerbed.length && n > 0; i++) {
            if (flowerbed[i] == 0 && plantable(flowerbed, i)) {
                flowerbed[i] = 1;
                n--;
            }
        }
        return (n == 0);
    }

    private boolean plantable(int[] flowerbed, int index) {
        boolean left = (index == 0) || (flowerbed[index - 1] == 0);
        boolean right = (index == flowerbed.length - 1) || (flowerbed[index + 1] == 0);
        return (left && right);
    }
}

File: 605-can-place-flowers/can-place-flowers.py
class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        flowerbed = [0] + flowerbed + [0]
        streak = 0
        flower = 0
        for spot in flowerbed:
            if spot == 0:
                streak += 1
            else:
                flower += (streak - 1) // 2
                streak = 0
        flower += (streak - 1) // 2
        return flower >= n